src/main/java/com/auction/retailService/service/StoreProductService.java

```diff
@@ -0,0 +1,115 @@
+package com.auction.retailService.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.auction.retailService.constant.ErrorMessageConstant;
+import com.auction.retailService.dto.ProductDTO;
+import com.auction.retailService.dto.StoreProductsDTO;
+import com.auction.retailService.entity.Product;
+import com.auction.retailService.entity.StoreProduct;
+import com.auction.retailService.entity.StoreProductBean;
+import com.auction.retailService.entity.StoreProductEmbeddable;
+import com.auction.retailService.exception.ProductDataException;
+import com.auction.retailService.exception.ProductNotFoundInStoreException;
+import com.auction.retailService.repository.StoreProductRepository;
+
+@Service
+public class StoreProductService {
+	
+	@Autowired
+	StoreProductRepository storeProductRepo;
+	
+	@Autowired
+	ProductService prodService;
+	
+	public StoreProduct updateStoreProduct(StoreProduct storeProduct) {
+		Optional<StoreProduct> storeProd =  storeProductRepo.findById(storeProduct.getStoreProductIdentity());
+		if(storeProd.isPresent())
+		{
+			storeProduct.setQuantity(storeProduct.getQuantity()+storeProd.get().getQuantity());
+		}
+		return storeProductRepo.save(storeProduct);
+	}
+	
+	public StoreProduct saveStoreProductData(StoreProduct storeProduct) {
+		return storeProductRepo.save(storeProduct);
+	}
+	
+	
+	public List<StoreProductBean> getStoreProducts(Long storeId) {
+		List<StoreProduct> storeProducts = storeProductRepo.findByStoreProductIdentityStoreId(storeId);
+		List<StoreProductBean> storeProductList = new ArrayList<StoreProductBean>();
+		for (StoreProduct store_prod : storeProducts) {
+			StoreProductBean bean = new StoreProductBean();
+			bean.setStoreId(store_prod.getStoreProductIdentity().getStoreId());
+			Optional<Product> prod = prodService.getProductById(store_prod.getStoreProductIdentity().getProductId());
+			Product product =  prod.isPresent()?prod.get():new Product();
+			bean.setProductId(product.getProductId());
```

Comment:
This is conversion logic. Segregate all conversion logic to a static utility class.